File: 2102-find-the-middle-index-in-array/find-the-middle-index-in-array.c
int findMiddleIndex(int* nums, int numsSize){
 int l_sum = 0, r_sum = 0;
    for (int i = 0; i < numsSize; ++i) {
        r_sum += nums[i];
    }

    for (int i = 0; i < numsSize; ++i) {
        if (l_sum == r_sum - nums[i]) {
            return i;
        }

        l_sum += nums[i];
        r_sum -= nums[i];
    }

    return -1;
}

File: 2102-find-the-middle-index-in-array/find-the-middle-index-in-array.py
class Solution:
    def findMiddleIndex(self, nums: List[int]) -> int:
        l_sum, r_sum =0, sum(nums)

        for i in range(len(nums)):
            if l_sum==r_sum-nums[i]:
                return i
            

            l_sum+=nums[i]
            r_sum-=nums[i]

        return -1